### PR TITLE
 Wrap operation about m_createdTransaction with mutex in some corner case

### PR DESCRIPTION
**Pull request by kaikawaliu**
### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libData/AccountData/AccountStore.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ void AccountStore::Init() {
 void AccountStore::InitSoft() {
   LOG_MARKER();
 
-  lock_guard<mutex> g(m_mutexPrimary);
+  unique_lock<shared_timed_mutex> g(m_mutexPrimary);
```
Comment (ShengguangXiao) [inline src/libData/AccountData/AccountStore.cpp]: Why change the mutex to shared_timed_mutex?

File: src/libData/AccountData/AccountStore.cpp [hunk 2]
```diff
@@ -85,11 +85,20 @@ AccountStore& AccountStore::GetInstance() {
   return accountstore;
 }
 
+bool AccountStore::Serialize(vector<unsigned char>& src,
+                             unsigned int offset) const {
+  LOG_MARKER();
+
+  shared_lock<shared_timed_mutex> lock(m_mutexPrimary);
```
Comment (ShengguangXiao) [inline src/libData/AccountData/AccountStore.cpp]: Why other places use unique_lock but here use shared_lock for m_mutexPrimary?

Comment (kaikawaliu) [reply src/libData/AccountData/AccountStore.cpp]: Because this is the only function we allow multithread accessing concurrently.
